impl/src/main/java/com/okta/sdk/impl/resource/AbstractPropertyRetriever.java

```diff
@@ -279,7 +279,15 @@ protected Map getMapProperty(String key) {
 
         if (value != null) {
             if (value instanceof String) {
-                return enumConverter.fromValue(type, value.toString());
+                E convertedValue;
+                try {
+                    convertedValue = enumConverter.fromValue(type, value.toString());
+                } catch (IllegalArgumentException e) {
+                    convertedValue = enumConverter.fromValue(type, "SDK_UNKNOWN");
+                    String msg = "Undeclared enum value {}.{}. Defaulting to SDK_UNKNOWN";
```

Comment:
I think the `key` is important here. The warning needs to instruct the developer on what to do, i.e. call `"get(" + key + ", String.class)"` (or whatever the actual method is)

Reply:
@bdemers updated